### PR TITLE
update to 0.5

### PR DESCRIPTION
**Pull request by davidvgalbraith**
@VladVega @demmer

### CODE REVIEW

File: lib/read.js [hunk 1]
```diff
@@ -1,45 +1,43 @@
...
-
-    initialize: function(options, params) {
+class ReadRedis extends AdapterRead {
+    constructor(options, params) {
+        super(options, params);
         if (!options.hasOwnProperty('raw')) {
             throw new Error('must specify a redis query with -raw');
         }
 
         this.command = options.raw.split(' ');
```
Comment (VladVega) [inline lib/read.js]: Proof that https://github.com/juttle/juttle/pull/457 is necessary.

Comment (VladVega) [inline lib/read.js]: Proof that juttle/juttle#457 is necessary.

File: lib/read.js [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
-    start: function() {
-        var self = this;
+    read(from, to, limit, state) {
         var keyword = this.command[0].toLowerCase();
 
         var transformation = transformations.get(keyword);
         if (!transformation) {
-            self.trigger('error', new Error('unsupported/invalid Redis command: ' + keyword));
-            return self.emit_eof();
+            throw new Error('unsupported/invalid Redis command: ' + keyword);
```
Comment (VladVega) [inline lib/read.js]: So I've noticed you're not into `this.runtimeError`. I'm curious why.

File: test/redis-adapter.spec.js [hunk 3]
```diff
@@ -1,29 +1,17 @@
...
 function _sort_by_value(list) {
     return _.sortBy(list, 'value');
 }
 
-Juttle.adapters.register(adapter.name, adapter);
-
-describe('redis source', function() {
-    this.timeout(300000);
-
+juttle_test_utils.withAdapterAPI(function() {
```
Comment (VladVega) [inline test/redis-adapter.spec.js]: Where is the adapter api used inside here?

**Comment by VladVega** [comment]
+1

File: lib/read.js (hunk 2 above)
Comment (davidvgalbraith) [inline lib/read.js]: `this.runtimeError` is great. I should use that.

File: test/redis-adapter.spec.js (hunk 3 above)
Comment (davidvgalbraith) [inline test/redis-adapter.spec.js]: Nowhere!

**Pull request merged**